consensus/consensus.go

```diff
@@ -145,4 +149,9 @@ type PoSA interface {
 	EnoughDistance(chain ChainReader, header *types.Header) bool
 	IsLocalBlock(header *types.Header) bool
 	AllowLightProcess(chain ChainReader, currentHeader *types.Header) bool
+	GetHighestJustifiedHeader(chain ChainHeaderReader, header *types.Header) *types.Header
+	GetHighestFinalizedNumber(chain ChainHeaderReader, header *types.Header) uint64
+	VerifyVote(chain ChainHeaderReader, vote *types.VoteEnvelope) bool
+	SetVotePool(votePool VotePool)
```

Comment:
It is a bit tricky to make `SetVotePool` as an interface.
<img width="549" alt="image" src="https://user-images.githubusercontent.com/90668111/167992072-e540cab7-7d8f-4022-ae67-d0861616ab74.png">

Can we make it a parameter when new the Parlia consensus?

Reply:
Because Parlia is a field in votePool, so to solve the import cycle, have to define an interface here

Reply:
As discussed, updated and removed this SetVotePool func